### ISSUE TITLE
Coversion is running very slowly

### ISSUE THREAD

**Issue by jcul** [openaudible/openaudible#1266]
Conversion seems to be running even slower than real time.

If the job is cancelled and restarted it starts quick, doing a few minutes of conversion a second.
But the job slows down so it is taking hours to convert an audiobook.
It's converting maybe 1 second every 3 seconds, so a 2 hour audiobook would take 6 hours to convert.

I cancelled any other conversion jobs so only 1 is running.

It is maxing out one CPU core.

ffmpeg command line:
```
/tmp/.mount_OpenAuCeNO4N/bin/linux_x86_64/ffmpeg -audible_iv [IV] -audible_key [KEY] -i [FILE].AAX -i [FILE.AAX].txt -map_metadata 1 -map_chapters 1 -map_metadata 0 -metadata:s:v title=Album cover -metadata:s:v comment=Cover (front) -codec:a libmp3lame -b:a 160k -id3v2_version 3 temp.mp3
```

Running on linux using the app image.

**Comment by jcul** [comment]
The AAX file it is converting right now is 68MB.

**Comment by openaudible** [comment]
That sounds very unusual. Maybe your disk is bad or it’s writing to a network disk in Timbuktu? Maybe check the log and see what the actual time to convert is. We list time, length and bytes after each conversion. 

From the docs: Conversion in progress? Converting a 500Mb 9 hour book to mp3 takes about 5 minutes, and about 20 seconds for m4b. Performance can be impacted by memory, CPU speed, and disk throughput. Watch the Library Status: Converting queue indicator, which might Converting 3 of 10. Click on the Converting line to see which books are converting and then select each book to see the state. Books that are converting will have a progress indicator under Audiobook Info: State that should show conversion go to 100%. Each successful conversion will turn the book color indicator green and decrement the queue. 

be sure it hasn’t queued multiple books and isn’t converting more than you think.

**Comment by jcul** [comment]
Here is the log from converting a 4 hour book.

```
Sep 19, 2023 5:15:44 P.M. org.openaudible.convert.ConvertJob convertFile
INFO: converting: The Order of Time to temp file: /home/jack/OpenAudible/tmp/BK_PAUK_001097UK_temp.mp3 from source /home/jack/OpenAudible/aax/The Order of Time.AAX length=125724481
Sep 19, 2023 5:15:44 P.M. org.openaudible.convert.ConvertJob convertFile
INFO: /tmp/.mount_OpenAuG8tIKt/bin/linux_x86_64/ffmpeg -audible_iv REDACTED -audible_key REDACTED -i "/home/jack/OpenAudible/aax/The Order of Time.AAX" -i "/home/jack/OpenAudible/tmp/The Order of Time.AAX.txt" -map_metadata 1 -map_chapters 1 -map_metadata 0 -metadata:s:v "title=Album cover" -metadata:s:v "comment=Cover (front)" -codec:a libmp3lame -b:a 160k -id3v2_version 3 /home/jack/OpenAudible/tmp/BK_PAUK_001097UK_temp.mp3 
Sep 20, 2023 7:19:27 A.M. org.openaudible.convert.ConvertJob convertFile
INFO: create MP3 /home/jack/OpenAudible/aax/The Order of Time.AAX result:0
Sep 20, 2023 7:19:27 A.M. org.openaudible.convert.ConvertJob convert
INFO: Converted The Order of Time to /home/jack/OpenAudible/books/The Order of Time.mp3 size=311516383 in 50622 seconds.
Sep 20, 2023 7:19:27 A.M. org.openaudible.Audible jobCompleted
INFO: ConvertQueue completed:The Order of Time ConvertQueue size:0 remain:0 success:1 fail:2 threads:1 jobs:0
Sep 20, 2023 7:19:29 A.M. org.openaudible.desktop.swt.manager.AudibleGUI _checkAutomation
INFO: Start auto conversion:1
```

You can see it took about 14 hours :)

**I think the issue is something to do with libmp3lame.**

Manually running the conversion it starts like this:

```
ffmpeg -audible_iv X -audible_key X -i "/home/jack/OpenAudible/aax/The Order of Time.AAX" -b:a 160k -id3v2_version 3 /tmp/test.mp3
...
frame=    0 fps=0.0 q=0.0 size=       1kB time=00:03:26.23 bitrate=   0.1kbits/s speed= 102x
```
But the speed keeps dropping, with a CPU maxed out.
After a few minutes it looks like this:

```
      encoder         : Lavc60.3.100 libmp3lame
frame=    0 fps=0.0 q=0.0 size=       1kB time=00:46:36.80 bitrate=   0.0kbits/s speed=6.37x
```

For comparison, running the conversion to aac, the rate stays at a constant ~26-28x.

```
ffmpeg -audible_iv X -audible_key X -i "/home/jack/OpenAudible/aax/The Order of Time.AAX" -b:a 160k -id3v2_version 3 /tmp/test.aac
...
  Stream #0:0(eng): Audio: aac (LC), 22050 Hz, stereo, fltp, 160 kb/s (default)
    Metadata:
      creation_time   : 2021-01-21T19:43:25.000000Z
      handler_name    : ?Apple Sound Media Handler
      vendor_id       : [0][0][0][0]
      encoder         : Lavc60.3.100 aac
size=   32512kB time=00:31:59.59 bitrate= 138.7kbits/s speed=28.5x
```

I'm running arch linux with lame version 3.100-4
```
$ pacman -Q --info lame
Name            : lame
Version         : 3.100-4
Description     : A high quality MPEG Audio Layer III (MP3) encoder
Architecture    : x86_64
URL             : http://lame.sourceforge.net/
Licenses        : LGPL
Groups          : None
Provides        : libmp3lame.so=0-64
Depends On      : ncurses
Optional Deps   : None
Required By     : ffmpeg  ffmpeg4.4  libsndfile  mencoder
Optional For    : None
Conflicts With  : None
Replaces        : None
Installed Size  : 1215.10 KiB
Packager        : David Runge <dvzrv@archlinux.org>
Build Date      : Sun 24 Jul 2022 00:05:49
Install Date    : Sun 29 Jan 2023 16:19:49
Install Reason  : Installed as a dependency for another package
Install Script  : No
Validated By    : Signature
```

I found this lame bug report: https://sourceforge.net/p/lame/bugs/491/
Looks like there was a performance issue, that was fixed in version 3.101.

Though it seems it is not released yet: https://lame.sourceforge.io/

**Comment by openaudible** [comment]
Good report. You can set the mp3 format to cbr in the OpenAudible preferences and see if that fixes it. We build the lame from scratch and I’ll check what version of lame we use. We need to upgrade them for the next version. I have arch on a vm and will try to reproduce. 

More soon.

**Comment by openaudible** [comment]
If you build from source, you'll need to replace our ffmpeg with yours unless you're just doing command line testing.

FYI, we build ffmpeg using the scripts here:
https://github.com/openaudible/ffmpeg-build/
which has a build-lame.sh script:
https://github.com/openaudible/ffmpeg-build/blob/main/build-lame.sh

Which gets the LAME source from: http://svn.code.sf.net/p/lame/svn/trunk/lame/   which is the latest (?): 3.101. Unless there is a newer, better branch we don't know about.

**Comment by openaudible** [comment]
Keep me posted. I wasn't able to find a good arch linux to test it with on a VM.. is there an ISO you can recommend with a desktop environment that would likely have the same problem?

**Comment by jcul** [comment]
OK my archlinux had a few packages out of date.
I first upgraded ffmpeg from 2:6.0-8 to 2:6.0-11, which made no difference.
Then I did a full system upgrade, and now locally the issue is resolved:

```
frame=    1 fps=0.0 q=-0.0 Lsize=  304204kB time=04:19:03.61 bitrate= 160.3kbits/s speed= 202x
video:614kB audio:303588kB subtitle:0kB other streams:0kB global headers:0kB muxing overhead: 0.000618%
```

I deleted OpenAudible's mp3s for this book and got it to reconvert.
It completed in about 3mins 30 sec, slower than running my local ffmpeg manually, but completely acceptable.

Honestly, I've no idea what could have caused this, maybe some library that got updated.
Happy to close this issue, unfortunately without really finding the root cause.
Maybe if someone else encounters the same thing, it will give them a hint.
